accepted/single-file/design.md

```diff
@@ -0,0 +1,281 @@
+# Single-file Publish
+
+Design for publishing apps as a single-file in .Net Core 3.0
+
+## Introduction
+
+The goal of this effort is enable .Net-Core apps to be published and distributed as a single executable.
+
+There are several strategies to implement this feature -- ranging from bundling the published files into zip file (ex: [Warp](https://github.com/dgiagio/warp)), to native compiling and linking all the binaries together (ex: [CoreRT](https://github.com/dotnet/corert)). These options, along with their cost/benefit analysis is explored in this [staging document](staging.md).
+
+#### Goals
+
+In .Net Core 3.0, we plan to implement a solution that 
+
+* Is widely compatible: Apps containing MSIL assemblies, ready-to-run assemblies, native binaries, configuration files, etc. can be packaged into one executable.
```

Comment:
I feel like configuration shouldn’t be included in this list by default. If I think about a common use case for this single-file system, then I am thinking of easy-to-distribute utility applications that likely come with a default configuration setup and can be further configured by a file that’s next to the application. If the configuration file gets embedded by default, then there needs to be additional logic to support an external configuration file (outside of the bundle).
But if the configuration file does not get embedded, this can be simply handled as the file being optional (this would be very easy to implement with M.E.Configuration).

Reply:
The problem is how you define "configuration". What is configuration for you may not be a configuration for me. As mentioned elsewhere in the doc, you will be able to customize what to include in the bundle.

Reply:
By configuration, here I mean .net-core's configuration files which are recognized and handled by the .net-core host/runtime. All other files are data-files from the point of view of the runtime, and should be handled as such.

Reply:
E.g. ASP.NET comes with bunch of its own configuration .json files that are just data-files from the point of the runtime.

I do not think the distinction between data files and configuration files matters much. All publishes stuff will get bundled (correct?) by default, it does not really matter what it is.

Reply:
Yes that's correct, Whatever is in the publish directory will get bundled, other files can be included too, as you mentioned above. Thanks.